File: 2.Het/1. Liskov helyettesítés sértése/LiskovSert.java
public class LiskovSert {
	
	static class Macska {
		public void szõrös() {}
	}
	
	static class Program {
		void fgv (Macska macska) {
			macska.szõrös();
		}
	}
	
	static class Perzsa extends Macska {}
	static class Szfinx extends Macska {}

	public static void main(String[] args) {
		Program program = new Program();
		Macska macska = new Macska();
		program.fgv(macska);
		
		Perzsa perzsa = new Perzsa();
		program.fgv(perzsa);
		
		Szfinx szfinx = new Szfinx();
		program.fgv(szfinx);
	}
}

File: 2.Het/1. Liskov helyettesítés sértése/LiskovSert.cpp
class Macska {
public:
	virtual void szoros() {};
};

class Program {
public:
	void fgv(Macska& macska) {
		macska.szoros();
	}
};

class Perzsa : public Macska {};
class Szfinx : public Macska {};

int main(int argc, char** argv) {
	Program program;
	Macska macska;
	program.fgv(macska);
	
	Perzsa perzsa;
	program.fgv(perzsa);
	
	Szfinx szfinx;
	program.fgv(szfinx);
}